packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx

```diff
@@ -610,6 +610,7 @@ const TinyMCE = (props: NoteBodyEditorProps, ref: any) => {
 				icons_url: 'gui/NoteEditor/NoteBody/TinyMCE/icons.js',
 				plugins: 'noneditable link joplinLists hr searchreplace codesample table',
 				noneditable_noneditable_class: 'joplin-editable', // Can be a regex too
+				link_context_toolbar: true,
```

Comment:
It seems that this isn't working with internal Joplin links — the toolbar is visible, but clicking "open link" does nothing.

Thank you for working on this!

Reply:
This makes this sort of a partial fix, while the regular hyperlinks are handled by an `onClick` event that scrolls the page to the element emitting the event. I couldnt find a similar fix while staying in TinyMCE scope only but maybe I have to put some more thought into it.

Reply:
Is there maybe an option that defines what TinyMCE considers a link? In which case the `joplin://` protocol could be added to the list

Reply:
Im pretty sure TinyMCE considers joplin links as links, as otherwise it wouldn't have opened the toolbar. Also I found that in terms of protocols, TinyMCE only allows setting a fallback protocol in case there is none supplied. The default fallback is set to https:// 
The issue here is mainly that the <a> has `target=blank` which tries to open a new blank browser tab inside electron. This is something that will require callback functions and platform specific configs to access the default browser instead of electron itself. 
At this point, modifying the link context toolbar itself to show a Ctrl + Click prompt seems like a better idea.

Reply:
I don't really understand the technical details without directly checking the code, but yes it needs to work for Joplin links too. Not sure what to suggest though. In any case it's either this or we close the PR since I'm not keen on half-working solutions

Reply:
Another option might be to override TinyMCE's "open selected link" action (perhaps with [code similar to how TinyMCE registers the button](https://github.com/tinymce/tinymce/blob/751e35f1419a6a060ded397dda1b2945bacaa711/modules/tinymce/src/plugins/link/main/ts/ui/Controls.ts#L25C33-L25C41)).

Reply:
I agree, it is better we drop this function for now, more so since Ctrl + Click is working well for most. 
Also, do you think this is a good idea for implementing localised tooltip inside TinyMCE? I follows the standard UI of other texteditors like LibreOffice Writer for prompting a Ctrl + Click.

Inside the `init()` function of TinyMCE,
```ts
editor.on('mouseover', (e:any) => {
	if(e.target.nodeName === 'A'){
		const existingTitle = e.target.getAttribute('title');
		if (existingTitle && 
                    !existingTitle.startsWith('Ctrl + Click to open')) {
			e.target.setAttribute('title', '(Ctrl + Click to open) ' + existingTitle);
		}
	}
})
```

Reply:
> Another option might be to override TinyMCE's "open selected link" button

Maybe it is possible to override the button to emulate Ctrl + click behaviour? The current anchor tag clicking behaviour implemented by TinyMCE tries to open the link in electron window itself and fails.

Reply:
> Also, do you think this is a good idea for implementing localised tooltip inside TinyMCE? It follows the standard UI of other texteditors like LibreOffice Writer for prompting a Ctrl + Click.

For the most part, that looks good to me!

Another option could be to add this functionality to Joplin's `packages/renderer` package. For example, in `linkReplacement.ts`:

https://github.com/laurent22/joplin/blob/c10e617870523e4734aee36ac90e4240496a0eaf/packages/renderer/MdToHtml/linkReplacement.ts#L112

It's currently possible to pass options to Joplin's renderer (e.g. `enableLongPress`).

Note, however that this may only show tooltips for existing links, and not for new links.

> Maybe it is possible to override the button to emulate Ctrl + click behaviour?

Joplin has an [`openItem`](https://github.com/laurent22/joplin/blob/dev/packages/app-desktop/gui/MainScreen/commands/openItem.ts) command that may be helpful here. It should be possible to execute commands with CommandService.execute ([example](https://github.com/laurent22/joplin/blob/c10e617870523e4734aee36ac90e4240496a0eaf/packages/app-desktop/gui/MainScreen/commands/newSubFolder.ts#L14)).

Reply:
> Another option could be to add this functionality to Joplin's `packages/renderer` package. For example, in `linkReplacement.ts`:
> 
> https://github.com/laurent22/joplin/blob/c10e617870523e4734aee36ac90e4240496a0eaf/packages/renderer/MdToHtml/linkReplacement.ts#L112
> 

This was the exact idea behind [my last PR](https://github.com/laurent22/joplin/pull/10215) (which I closed), because the `linkreplacement.ts` package was used in more places than the feature is required.